### PR TITLE
Forgot Password Flow (w/PopUp Modal & KeyboardAvoidingView)

### PR DESCRIPTION
**Pull request by aksharans**
### Administrative Info
Monday Board ID: 853642866
Make sure your branch name conforms to: `<feature/staging/hotfix/...>/[username]/[Monday Item ID]-[3-4 word description separated by dashes]`. Otherwise, please rename your branch and create a new PR.

### Changes
What changes did you make?
- AuthContext: implemented setPasswordResetEmail()
- ForgotPasswordScreeen: text input to send password reset email, with error handling and popup modal on success
- PopUpModal: modal to indicate success and navigate to previous screen in stack
- LoginScreen & SignUpScreen: added KeyboardAvoidingView

-included some commented code, but can remove before merging

### Testing
How did you confirm your changes worked? 
- ran on staging server, added valid & invalid emails and successfully signed in with resetted pswd

### Confirmation of Change 
Upload a screenshot, if possible. Otherwise, please provide instructions on how to see the change.
npm run staging, click Forgot Password in Sign In Screen, valid emails in firebase

### CODE REVIEW

File: mobile/src/context/AuthContext.tsx [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import React, { createContext, useContext, useEffect, useMemo, useState } from 'react';
+// import { Alert } from 'react-native';
```
Comment (dacruz21) [inline mobile/src/context/AuthContext.tsx:2]: remove comments please

File: mobile/src/context/AuthContext.tsx [hunk 2]
```diff
@@ -121,6 +124,40 @@ export const AuthProvider: React.FC = ({ children }) => {
...
+      bundleId: 'com.example.ios'
+    },
+    android: {
+      packageName: 'com.example.android',
+      installApp: true,
+      minimumVersion: '12'
+    },
+    handleCodeInApp: true
+  };
+  */
```
Comment (dacruz21) [inline mobile/src/context/AuthContext.tsx:140]: ^

File: mobile/src/context/AuthContext.tsx [hunk 3]
```diff
@@ -121,6 +124,40 @@ export const AuthProvider: React.FC = ({ children }) => {
...
+    
+    (async () => {
+      try {
+        const res: string[] = await auth.fetchSignInMethodsForEmail(email);
+        if (res.length != 0) {
+          auth.sendPasswordResetEmail(email);
+          // Callback to set modal visibility to true (indicate success)
+          showModal();
+        } else {
+          setError(new Error('User With Email Does Not Exist'));
```
Comment (dacruz21) [inline mobile/src/context/AuthContext.tsx:152]: How about "Could not find an account with that email address"?

**Review by typokign — CHANGES_REQUESTED** [review]
Great work, thank you both!!

Some thoughts:
* I wasn't able to get a good `npm install` on your branch, JS rot lol. I opened PR #53 to fix the install errors. CI seems happy for some reason so we can land these in either order
* KeyboardAvoidingView still doesn't work on Android :( Oh well, if it works on iOS it's definitely better than nothing, thanks for making the change
* Let's add some explanation text to the reset password form, something like "Enter the email address you used when creating your account to receive instructions for resetting your password", idk something like that. Then we can just change the "Send Email to Reset Password" button to just Reset or Send or something
* You can add a prop on the TextInput to specify that the expected value is an email address - this is pretty useful as it changes the layout of the keyboard on the device, moves the @ sign to the main keyboard, adds autofills for gmail.com, yahoo.com, etc.
* I like the modal :D

**Review by typokign — APPROVED** [review]
Updated the wording after talking to Robyn today, LGTM!

File: mobile/src/screens/login/LoginScreen.tsx [hunk 4]
```diff
@@ -26,37 +26,54 @@ export const LoginScreen: React.FC = () => {
...
+
+          <View style={styles.textContainer}>
+              
+            <TouchableOpacity style={styles.rememberMeContainer} onPress={() => setRememberMe(!rememberMe)}>
+              <Checkbox value={rememberMe} onChange={setRememberMe} inverted />
+              <Text style={styles.rememberMeText}>{i18n.t('rememberMe')}</Text>
+            </TouchableOpacity>
+
+            <TouchableOpacity style={styles.forgotPswdContainer} onPress={() => navigation.navigate("ForgotPswd")}>
+              <Text style={styles.forgotPasswordText}>{i18n.t('forgotPassword')} </Text>
```
Comment (petabite) [inline mobile/src/screens/login/LoginScreen.tsx:64]: Maybe change the double quotes to single quotes (also in a couple of other places). I believe David is working on the linter though, so probably not a big issue.

File: mobile/src/screens/login/ForgotPasswordScreen.tsx [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+          <Image source={require('../../../assets/images/logo-white.png')} style={styles.logo} />
+        </View>
+        
+        <PopUpModal text={i18n.t('passwordResetEmailSent')} setModalVisible={setModalVisible} modalVisible={modalVisible} />
+
+
+        <View style={styles.container}>  
+          <Text style={[TextStyles.heading1, styles.instructions]}>{i18n.t("forgotPassword")}</Text>
+          <Text style={[TextStyles.caption2, styles.instructions]}>{i18n.t('resetPasswordInstructions')}</Text>
+          <TextInput style={[styles.input, TextStyles.caption3]} value={email} onChangeText={setEmail} placeholder={i18n.t('email')} placeholderTextColor={Colors.gray} textContentType="emailAddress" keyboardType="email-address"/>
```
Comment (petabite) [inline mobile/src/screens/login/ForgotPasswordScreen.tsx:67]:
```suggestion
          <TextInput style={[styles.input, TextStyles.caption3]} value={email} onChangeText={setEmail} placeholder={i18n.t('email')} placeholderTextColor={Colors.gray} textContentType="emailAddress" keyboardType="email-address" autoCapitalize='none'/>
```
Super small UX issue - I noticed that the email input automatically capitalizes the first character, which is a little weird. Adding this should fix that.

**Review by petabite — APPROVED** [review]
LGTM! Left some small suggestions below if you want to add them in.

**Pull request merged**